src/metabase/driver/sql_jdbc/sync/describe_table.clj

```diff
@@ -264,14 +264,25 @@
 (defn describe-table-fks
   "Default implementation of [[metabase.driver/describe-table-fks]] for SQL JDBC drivers. Uses JDBC DatabaseMetaData."
   [driver db-or-id-or-spec-or-conn table & [db-name-or-nil]]
-  (if (instance? Connection db-or-id-or-spec-or-conn)
-    (describe-table-fks* driver db-or-id-or-spec-or-conn table db-name-or-nil)
-    (sql-jdbc.execute/do-with-connection-with-options
-     driver
-     db-or-id-or-spec-or-conn
-     nil
-     (fn [^Connection conn]
-       (describe-table-fks* driver conn table db-name-or-nil)))))
+  (sql-jdbc.execute/do-with-connection-with-options
+   driver
+   db-or-id-or-spec-or-conn
```

Comment:
`do-with-connection-with-options` does not take `conn` as an argument.

this will fail

```clojure
(let [db (toucan2.core/select-one :model/Database :engine :h2)]
  (sql-jdbc.execute/do-with-connection-with-options
   :h2
   db
   nil
   (fn [conn1]
     (sql-jdbc.execute/do-with-connection-with-options
      :h2
      conn1
      nil
      (fn [conn2]
        (jdbc/query {:connection conn2} "select 1;"))))))
```

But I doubt we use `defn describe-table-fks` with a connection anywhere. Still, we need to check to make sure we don't create regression here.